gson/src/main/java/com/google/gson/internal/reflect/ReflectionHelper.java

```diff
@@ -65,10 +82,131 @@ public static String tryMakeAccessible(Constructor<?> constructor) {
     }
   }
 
-  public static RuntimeException createExceptionForUnexpectedIllegalAccess(IllegalAccessException exception) {
+  public static boolean isRecord(Class<?> raw) {
+    return RECORD_HELPER.isRecord(raw);
+  }
+
+  public static String[] recordFields(Class<?> raw) {
+    return RECORD_HELPER.recordFields(raw);
```

Comment:
Maybe both methods should be renamed to `getComponentNames`.

Reply:
Done